File: FoodObfuscator/src/nz/ac/auckland/cs/FoodObfuscator.java
package nz.ac.auckland.cs;

import java.io.BufferedReader;
import java.io.FileNotFoundException;
import java.io.FileReader;

public class FoodObfuscator {

    public static void main(String[] args) {
        if(args.length != 1) {
            System.err.println("Expected 1 argument of input file name, got " + args.length);
            System.exit(1);
        }

        try {
            BufferedReader s = new BufferedReader(new FileReader(args[0]));
            // read file here or do stuff
        } catch (FileNotFoundException e) {
            System.err.println("File not found");
            System.exit(1);
        }
    }

}

File: FoodObfuscator/src/nz/ac/auckland/cs/FoodObfuscator.kt
package nz.ac.auckland.cs

import java.io.BufferedReader
import java.io.FileReader
import java.util.stream.Stream

fun main(args: Array<String>) {
    if (args.size != 1) System.exit(1)
    val bufferedReader = BufferedReader(FileReader(args[0]))
    val text: Stream<String?> = bufferedReader.lines()
    text.filter{x -> x != null}.forEach {
        // do stuff
    }
}